docs/examples/distributed/003_multi_node/main.py

```diff
@@ -0,0 +1,279 @@
+"""Multi-GPU Training example."""
+import logging
+import os
+
+import torch
+import torch.distributed
+from torch import Tensor, nn
+from torch.nn import functional as F
+from torch.utils.data import DataLoader, random_split
+from torch.utils.data.distributed import DistributedSampler
+from torchvision import transforms
+from torchvision.datasets import CIFAR10
+from torchvision.models import resnet18
+from tqdm import tqdm
+
+
+def main():
+    training_epochs = 10
+    learning_rate = 5e-4
+    weight_decay = 1e-4
+    batch_size = 128  # NOTE: This is the "local" batch size, per-GPU.
+
+    # Check that the GPU is available
+    assert torch.cuda.is_available() and torch.cuda.device_count() > 0
+    rank, world_size, local_rank = setup()
+    is_master = rank == 0
+    is_local_master = local_rank == 0
+    device = torch.device("cuda", local_rank)
+    import rich.logging
```

Comment:
Was there some issues with having `import rich.logging` at the top like in 002?

Reply:
Fixed in c23e81e31de1618db4aa0930608fd3aa6e9b3e8f